lib/http.go

```diff
@@ -112,7 +112,19 @@ func (c HTTPClient) do(ctx context.Context, addr string, httpMethod string, mime
 	var err error
 
 	if httpMethod == http.MethodGet || httpMethod == http.MethodDelete {
-		req, err = http.NewRequest(httpMethod, addr, nil)
+		qvars := ""
+		if params != nil {
+			if pm, ok := params.(map[string]string); ok {
+				for k, v := range pm {
+					if qvars == "" {
+						qvars = fmt.Sprintf("?%s=%s", k, v)
```

Comment:
I think `k` and `v` need to be URL escaped here?

Reply:
```go
               u, err := url.Parse(addr)
               if err != nil {
                 return err
               }
               
		if pm, ok := params.(map[string]string); ok {
			qvars := u.Query()
			for k, v := range pm {
				qvars.Set(k, v)
			}
			u.RawQuery = qvars.Encode()
		}
		req, err = http.NewRequest(httpMethod, u, nil)
```